geopandas/tools/clip.py

```diff
@@ -182,7 +182,7 @@ def clip(gdf, mask, keep_geom_type=False):
             elif new_collection or more_types:
                 orig_type = gdf.geom_type.iloc[0]
                 if new_collection:
-                    clipped = clipped.explode()
+                    clipped = clipped.explode(index_parts=True)
```

Comment:
This is the backwards compatible change, but, looking at this code now, we should actually set this to False? (we directly return this object, so which gets a MultiIndex in this case)